extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantContextFactory.java

```diff
@@ -214,12 +213,12 @@ private Uni<TenantConfigContext> createTenantContext(OidcTenantConfig oidcTenant
         }
 
         if (oidcConfig.roles().source().orElse(null) == io.quarkus.oidc.runtime.OidcTenantConfig.Roles.Source.userinfo
-                && !enableUserInfo(oidcConfig)) {
```

Comment:
This method actually has a side-effect of setting it to `true` in one of the branches

Reply:
this side-effect is not necessary anymore

Reply:
AKA, we detect this sooner, that's all.

Reply:
@michalvavrik Sorry, I think what is here now is correct, that `enableUserInfo` is only reacting to the detected injection point, if the user requested not to use UserInfo than the user decision wins

Reply:
@michalvavrik I was not disagreeing, and confirming I was good with this change